### ISSUE TITLE
Not working on Android 14  please update the package

### ISSUE THREAD

**Issue by akhileshgoswami** [Dev-hwang/flutter_activity_recognition#5]
Not working on Android 14  please update the package

**Comment by Rumbel** [comment]
Hello @akhileshgoswami, I too ran into some problems with the package and wasnt sure if it has something to do with the OS version. Could you please elaborate the problem that you ran into a litte bit so i can see if  i am right in this Issue?
Thank you.

**Comment by shirishwaytoweb** [comment]
release APK not getting activity on android 14 any solution for that

**Comment by thorito** [comment]
I'm using android 14 and it doesn't work.

Any ideas?

**Comment by thorito** [comment]
I'm using android 14 and it doesn't work.

Any ideas?

**Comment by fuadarradhi** [comment]
any update on this @Dev-hwang

**Comment by Dev-hwang** [comment]
upgrade to the latest version.

and, set targetSdkVersion to 34 in app/build.gradle.

```gradle
defaultConfig {
    applicationId "com.pravera.flutter_activity_recognition_example"
    minSdkVersion 21
    targetSdkVersion 34
    versionCode flutterVersionCode.toInteger()
    versionName flutterVersionName
}
```

**Comment by albright999** [comment]
Tried 

minSdkVersion 21
targetSdkVersion 34

Android builds and runs, but I only seem to get activity of UNKNOWN, LOW confidence   (Pixel 8)
It builds and runs on iOS, recognition seems very good.

**Comment by timsar2** [comment]
I use it with foreground service 
What activity recognition stop working after six hours or maybe during the night. 
The application has battery restrict permission
The foreground service is type of health

**Comment by thorito** [comment]
I have a Pixel 7 Pro (Android 15)  and I get 'UNKNOWN' quite often.
When I'm driving it changes the mode to 'IN_VEHICLE' but when I stop it usually doesn't detect the 'STILL' event

I'm using flutter_background_service to keep the running in the background

Any idea what's going on?

**Comment by timsar2** [comment]
Maybe it's going off for timeout. new foreground service has type and timeout duration.
this must call stop stream when timeout call back fire from android os

**Comment by timsar2** [comment]
> I'm using android 14 and it doesn't work.
> 
> Any ideas?

I see you are making your own plugin, dose it work on android14?
 [https://pub.dev/packages/custom_activity_recognition](https://pub.dev/packages/custom_activity_recognition)

I use foreground service to stay up with activity recognition, but during the night, when the app is not using, the os will kill the service and when I open the app again, the activity recognition stop working.

**Comment by thorito** [comment]
> > I'm using android 14 and it doesn't work.
> > 
> > Any ideas?
> 
> I see you are making your own plugin, dose it work on android14?
>  [https://pub.dev/packages/custom_activity_recognition](https://pub.dev/packages/custom_activity_recognition)
> 
> I use foreground service to stay up with activity recognition, but during the night, when the app is not using, the os will kill the service and when I open the app again, the activity recognition stop working.
> 
> 

You are right. I guess if I add a foreground service it should stick around. I can confirm that the recognition works on Android 15 in the foreground.
